### ISSUE TITLE
[Usage]: How to reproduce the result with less than 8*A100 (also hope to reproduce the training dynamic)

### ISSUE THREAD

**Issue by yfpeng1234** [RLinf/RLinf#563]
### Problem description

Dear Author,
Thanks for providing this awesome framework. I suppose that most of your experiments in RLinf, Pi_RL papers are conducted on 8*H100. I'm interested in how to reproduce both the final success rate and the training dynamic on 4*A100. I'm not sure how to adjust the following arguments accordingly:
1. total_num_envs: should total_num_envs*rollout_epoch be the same as yours, in my case, I half the total_num_envs, then should I double rollout_epoch
2. micro_batch_size and global_batch_size keeps unchanged?
3. lr and updata_epoch?
By the way, does "RLtrainsteps" in Table8 of pi_RL refers to "max_epochs" in the configuration file? In that case, a few hundreds rather than 1k is enough?

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by chenkang455** [comment]
Hi @yfpeng1234, if your A100 has 80GB memory, you should be able to run experiments directly with the default configuration. Otherwise, please refer to the [Minimum Test Case](https://rlinf.readthedocs.io/en/latest/rst_source/examples/pi0.html#:~:text=%E2%AD%90%202.4%20Minimum%20Test%20Case%20%E2%AD%90) section, which is optimized for lower GPU memory usage. Note that, based on my experiments, the final performance may be slightly lower compared to the full configuration.

Regarding your questions:

1. Correct. Reducing num-envs while increasing rollout-epoch lowers memory consumption, decreasing the number of concurrent environments, though this comes at the cost of increased training time.

2. If you encounter OOM errors, you can decrease the ``micro_batch_size`` while keeping the global batch size constant. However, be aware that an excessively small micro-batch size might affect training stability (a behavior we are currently investigating in [Issue #537](https://github.com/RLinf/RLinf/issues/537)).

3. Yes, in most cases, we stop training around step 400 rather than running the full 1000 steps. You can determine the optimal stopping point by monitoring the convergence curves. For reference, we typically stop at step 450 for MetaWorld and step 100 for CALVIN.

**Comment by yfpeng1234** [comment]
Thanks so much for your prompt reply, Kang!

**Comment by yfpeng1234** [comment]
Hi Kang @chenkang455 , May I ask another question regarding VLM LoRA finetune? Most of configurations freeze VLM and only finetune action expert. As stated in pi_RL, it requires a more conservative learning rate and update_epoch. Is there any intuition on how to adjust LR and update_epoch if I want to finetune the VLM simultaneously? By the way, I also observed that LR for maniskill experiment is somehow intricate. Does that mean maniskill experiment is sensitive to LR?

**Comment by chenkang455** [comment]
@yfpeng1234 If you are fine-tuning a VLM simultaneously, the increased model parameters typically necessitate a more conservative lr and update-epoch. I recommend starting with the default settings and monitoring for training instability. Specifically, keep an eye on whether the approx-kl significantly exceeds 0.02, which is generally considered the threshold for a stable update. If it does, consider reducing both the LR and the update-epoch.

Regarding the learning rate for ManiSkill, based on our empirical results, using an LR of 5e-6 and an update-epoch of 4 (the same hyperparameters used for CALVIN and MetaWorld) is a solid baseline for most experiments.

**Comment by yfpeng1234** [comment]
Hi @chenkang455 , sorry to bother you. I still have problem to reproduce Table3.
I run "bash examples/embodiment/run_embodiment.sh maniskill_ppo_openpi_pi05" to train pi_0.5.
The only modification I did is to change max_epochs from 1000 to 150, as suggested in Table8(hyper-parameters).
Then I run "bash examples/embodiment/eval_mani_ood.sh" to evaluate the model.
Here I change total_num_envs from 16 to 256, as suggested in another issue for more stable result.
The result I got:

<img width="390" height="449" alt="Image" src="https://github.com/user-attachments/assets/2d6a0ed3-bfc6-46c1-ac8d-2659ed0a8e8e" />
It seems that IND metric can not meet the value reported in paper (71.8<->89.7), IND even lower than vision OOD.
Should I further train model for more steps or evaluate for more envs?
Best

**Comment by Tonghe-Zhang** [comment]
Hi @yfpeng1234 , thank you for your interest in our work! We are happy to see that our algorithm is making improvement in your setup. About the discrepancy, according to our [paper](https://arxiv.org/pdf/2510.25889), we run the training in 320 parallel environments for 150 episodes, and each episode contains 48 steps. Our configuration may consume more samples than your training setup. Our evaluation is also conducted in 320 parallel environments. Under your setting, the model's performance could be a little bit worse than our results due to a lack of samples, and I assume that training longer in your setting may help.
